common/src/IO/LoadTextureCollection.cpp

```diff
@@ -88,8 +90,33 @@ Result<Assets::Texture, ReadTextureError> readTexture(
   const size_t prefixLength,
   const std::optional<Assets::Palette>& palette)
 {
-  static const auto imageFileExtensions =
-    std::vector<std::string>{".jpg", ".jpeg", ".png", ".tga", ".bmp"};
+  static std::vector<std::string> imageFileExtensions;
+  if (imageFileExtensions.empty())
+  {
+    const int count = FreeImage_GetFIFCount();
+    assert(count >= 0);
+    imageFileExtensions.reserve(static_cast<size_t>(count));
+    for (int i = 0; i < count; ++i)
+    {
+      auto fif = static_cast<FREE_IMAGE_FORMAT>(i);
+      if (FreeImage_IsPluginEnabled(fif))
+      {
+        char const* extensionList = FreeImage_GetFIFExtensionList(fif);
+        size_t extensionListLen = strlen(extensionList) + 1;
```

Comment:
Please don't use C strings - you should immediately convert them to `std::string` and then only work with that.